db/snapshot/backup_task.cc

```diff
@@ -55,17 +55,29 @@ tasks::is_abortable backup_task_impl::is_abortable() const noexcept {
 }
 
 future<tasks::task_manager::task::progress> backup_task_impl::get_progress() const {
-    co_return tasks::task_manager::task::progress {
-        .completed = _progress.uploaded,
-        .total = _progress.total,
+    struct adder {
+        tasks::task_manager::task::progress result;
+        future<> operator()(s3::upload_progress p) {
+            result.completed += p.uploaded;
+            result.total += p.total;
+            return make_ready_future<>();
+        }
+        tasks::task_manager::task::progress get() const { return result; }
+    };
+    auto p = co_await _snap_ctl.container().map_reduce(adder{}, [this](auto&) -> s3::upload_progress {
+        return _progress_per_shard[this_shard_id()];
```

Comment:
map_reduce gives you the sharded object so you don't have to play these games.

Reply:
I recommend using the map_reduce0 variant with
 - map: extract the shard's progress from the sharded object, as an object that supports operator+
 - initial: the same object with an empty initializer
 - reduce: std::plus<>()

Reply:
Hope I got it right